File: sparkcql-core/src/main/scala/gov/va/sparkcql/adapter/library/FileLibraryAdapterFactory.scala
package gov.va.sparkcql.adapter.library

import gov.va.sparkcql.di.ComponentConfiguration

class FileLibraryAdapterFactory extends LibraryAdapterFactory {

  def create(configuration: ComponentConfiguration): LibraryAdapter = {
    new FileLibraryAdapter(configuration.read("sparkcql.filelibraryadapter.path", "./"))
  }
}

File: sparkcql-core/src/main/java/gov/va/sparkcql/adapter/library/FileLibraryAdapterFactory.java
package gov.va.sparkcql.adapter.library;

import org.apache.spark.sql.SparkSession;

import gov.va.sparkcql.Configuration;

public class FileLibraryAdapterFactory implements LibraryAdapterFactory {

    @Override
    public LibraryAdapter create(Configuration configuration, SparkSession spark) {
        return new FileLibraryAdapter(configuration.read("sparkcql.filelibraryadapter.path", null));
    }
}